osu.Desktop/Program.cs

```diff
@@ -101,12 +93,56 @@ public static void Main(string[] args)
                     {
                         Logger.Error(ex, "Failed to start legacy IPC provider");
                     }
+
+                    Logger.Log("Starting IPC handler...");
+                    ipc = new IpcServer(host);
```

Comment:
Before I even begin to review this, can you explain why you are making your own `IpcServer` rather than using the proposed existing flow provided by `GameHost`? Is there something it doesn't provide that is required here?

https://github.com/ppy/osu-framework/blob/master/osu.Framework/Platform/GameHost.cs#L96-L119